docs/TuxArchitecture.md

```diff
@@ -0,0 +1,186 @@
+    architect(unreadStore).itNeeds('unreadCount').and('users');
+```
+
+Lastly, you have the option to directly need stores if you do not want to use this string based syntax:
+
+```
+    architect(roomStore).itNeeds(userStore);
+    architect(threadStore).itNeeds(userStore, roomStore);
+    architect(messageStore).itNeeds(userStore, threadStore);
+    architect(privateMessageStore).itNeeds(userStore);
+    architect(unreadStore).itNeeds(roomStore, threadStore, messageStore, privateMessageStore);
+```
+
+While this approach is shorter it can be advantageous to use the string approach as it allows you to write your dependencies in terms of the actual outputs of your stores and thus provides self-documenting code that can make it easier to reason about your application architecture down the line.
+
+Below is the full API documentation for the `architect` method.
+
+####<a id="Architecture-architect"></a>Architecture.architect [#](#Architecture-architect)
+Accepts a store and returns an `architectureChain` instance to define the inputs and outputs for the store.
+
+```
+    var architectureChain = Architecture.architect(storeToArchitect);
+```
+
+#####<a id="architect-storeToArchitect"></a>Parameter - `storeToArchitect` - type: OBJECT - required [#](#architect-storeToArchitect)
+Store to architect. `itNeeds` will add dependencies to this store, `itOutputs` will define the outputs that other stores can use to add a dependency for this store via `itNeeds`.
+
+```
+    var storeToArchitect = require('../stores/userStore');
+```
+
+#####<a id="architect-architectureChain"></a>Return - `architectureChain` - type: OBJECT [#](#architect-architectureChain)
+Object to chain architecture methods onto. Defined below.
+
+####<a id="architectureChain"></a>architectureChain - type: OBJECT [#](#architectureChain)
+The `architectureChain` instance returned from the `Architecture.architect` method. It possesses the following methods, each of which return this same architectureChain instance in order to allow method chaining:
+
+#####<a id="architectureChain-itNeeds"></a>Method - `architectureChain.itNeeds` - type: FUNCTION [#](#architectureChain-itNeeds)
+Accepts any number of inputs: stores, output strings, or arrays of stores and/or output strings. Adds a dependency to the `storeToArchitect` for each store mapped to the passed in inputs. Returns this `architectureChain` instance for continued method chaining.
+
+```
+    architectureChain.itNeeds(storeOrInput, additionalStoresOrInputs...);
+```
+
+######<a id="itNeeds-storeOrInput"></a>Parameter - `storeOrInput` - type: STRING, OBJECT, ARRAY of STRINGs and/or OBJECTs - required [#](#itNeeds-storeOrInput)
+Adds a dependency for the store or stores that map to the passed in string. If a store is passed adds a dependency for the passed in store. If an array is passed in adds a dependency for each element in the passed in array. Will do this for each input.
```

Comment:
Recommend changing the second sentence to be something like "If a store is passed in as an input, it adds..."